### PR TITLE
Fix bug in build-with-buck with library targets

### PR DESCRIPTION
**Pull request by adamhowardprice**
Summary: Running `buck project //src:foo --build-with-buck` creates a project that cannot be built in Xcode if //src:foo is a library target. Previously we used one shell script that 1) built the project with Buck and 2) fixed UUIDs in the binary. Now we add the shell script to build in all cases, but only add the fix_uuid script if it is an apple_bundle.

Test plan: Unit tests updated. `ant java-test -Dtest.class=ProjectGeneratorTest`. Also, build an `apple_bundle`, `apple_binary` and `apple_library`, with `buck project <target> --build-with-buck --combined-project` and make sure all generated Xcode projects build correctly.

### CODE REVIEW

File: test/com/facebook/buck/apple/ProjectGeneratorTest.java [hunk 1]
```diff
@@ -826,7 +826,7 @@ public void testHalideLibraryRule() throws IOException {
 
     ProjectGenerator projectGenerator =
       createProjectGeneratorForCombinedProject(
-        ImmutableSet.<TargetNode<?>>of(compiler, lib));
+          ImmutableSet.<TargetNode<?>>of(compiler, lib));
```
Comment (beefon) [inline test/com/facebook/buck/apple/ProjectGeneratorTest.java]: Please revert these changes

File: test/com/facebook/buck/apple/ProjectGeneratorTest.java [hunk 2]
```diff
@@ -816,9 +816,9 @@ public void testHalideLibraryRule() throws IOException {
       .build();
     TargetNode<?> compiler = new HalideLibraryBuilder(compilerTarget)
       .setSrcs(
-        ImmutableSortedSet.of(
-          SourceWithFlags.of(new TestSourcePath("main.cpp")),
-          SourceWithFlags.of(new TestSourcePath("filter.cpp"))))
+          ImmutableSortedSet.of(
+              SourceWithFlags.of(new TestSourcePath("main.cpp")),
+              SourceWithFlags.of(new TestSourcePath("filter.cpp"))))
```
Comment (beefon) [inline test/com/facebook/buck/apple/ProjectGeneratorTest.java]: Please revert these changes

File: src/com/facebook/buck/apple/ProjectGenerator.java [hunk 3]
```diff
@@ -137,6 +139,7 @@
 
   private static final Logger LOG = Logger.get(ProjectGenerator.class);
   private static final String BUILD_WITH_BUCK_TEMPLATE = "build-with-buck.st";
+  private static final String BUILD_WITH_BUCK_FIX_UUID_TEMPLATE = "build-with-buck-fix-uuid.st";
```
Comment (beefon) [inline src/com/facebook/buck/apple/ProjectGenerator.java]: I think this should be renamed to `fix-uuid.st` as this script does not actually build anything.

File: src/com/facebook/buck/apple/ProjectGenerator.java [hunk 4]
```diff
@@ -648,7 +691,7 @@ private static Path getHalideOutputPath(BuildTarget target) {
     PBXShellScriptBuildPhase scriptPhase = new PBXShellScriptBuildPhase();
     ST template = new ST(
       Resources.toString(
-        Resources.getResource(ProjectGenerator.class, HALIDE_COMPILER_TEMPLATE),
+        getResource(ProjectGenerator.class, HALIDE_COMPILER_TEMPLATE),
```
Comment (beefon) [inline src/com/facebook/buck/apple/ProjectGenerator.java]: I think you can also revert this.

File: test/com/facebook/buck/apple/ProjectGeneratorTest.java (hunk 2 above)
Comment (adamhowardprice) [inline test/com/facebook/buck/apple/ProjectGeneratorTest.java]: Done

File: test/com/facebook/buck/apple/ProjectGeneratorTest.java (hunk 1 above)
Comment (adamhowardprice) [inline test/com/facebook/buck/apple/ProjectGeneratorTest.java]: Done

File: src/com/facebook/buck/apple/ProjectGenerator.java (hunk 4 above)
Comment (adamhowardprice) [inline src/com/facebook/buck/apple/ProjectGenerator.java]: Done

File: src/com/facebook/buck/apple/ProjectGenerator.java [hunk 5]
```diff
@@ -554,6 +519,83 @@ private String getBuildFlags() {
...
+  }
+
+  private String getFixUUIDShellScript(TargetNode<?> targetNode) {
+    ST template;
+    try {
+      template = new ST(Resources.toString(
+          getResource(ProjectGenerator.class, FIX_UUID_TEMPLATE), Charsets.UTF_8));
+    } catch (IOException e) {
+      throw new RuntimeException(
+          "There was an error loading 'fix-uuid.st' template", e);
```
Comment (beefon) [inline src/com/facebook/buck/apple/ProjectGenerator.java]: nit: replace `fix-uuid.st` with `FIX_UUID_TEMPLATE` const

File: src/com/facebook/buck/apple/ProjectGenerator.java [hunk 6]
```diff
@@ -554,6 +519,83 @@ private String getBuildFlags() {
...
   }
 
+  private String getBuildWithBuckShellScript(TargetNode<?> targetNode) {
+    ST template;
+    try {
+      template = new ST(Resources.toString(
+          getResource(ProjectGenerator.class, BUILD_WITH_BUCK_TEMPLATE),
+          Charsets.UTF_8));
+    } catch (IOException e) {
+      throw new RuntimeException("There was an error loading 'build-with-buck.st' template", e);
```
Comment (beefon) [inline src/com/facebook/buck/apple/ProjectGenerator.java]: nit: replace 'build-with-buck.st' with BUILD_WITH_BUCK_TEMPLATE const

File: src/com/facebook/buck/apple/ProjectGenerator.java (hunk 5 above)
Comment (adamhowardprice) [inline src/com/facebook/buck/apple/ProjectGenerator.java]: :heavy_check_mark:

File: src/com/facebook/buck/apple/ProjectGenerator.java (hunk 6 above)
Comment (adamhowardprice) [inline src/com/facebook/buck/apple/ProjectGenerator.java]: :heavy_check_mark:

**Comment by beefon** [comment]
@facebook-github-bot shipit

**Comment by adamhowardprice** [comment]
@beefon I'm not sure how Facebook's internal bot handles situations like this. When you tell the bot "ship it", can you still append more commits on top? I found some bugs in the tests, which were being skipped previously, so I added a commit to fix that. Can you review that commit and import it too? Or should I just open a new pull request once this one gets merged?

**Comment by sdwilsh** [comment]
We'll have to do a separate PR.  I'm just about to push the scrubbed changes out; this was landed in our internal tree yesterday morning, alas.

It should be easy to do a PR though once you rebase :)
